### PR TITLE
Issue 2188 - docker login credentials in the OS keyring

### PR DESCRIPTION
**Pull request by sirotsinskuy**
Signed-off-by: Antonii Sirotsinskyi <asiro@softserveinc.com>

### CODE REVIEW

File: cli/cliutils/cliutils.go [hunk 1]
```diff
@@ -214,6 +215,49 @@ func GetDockerAuth(domain string) (auth dockerclient.AuthConfiguration, err erro
 		}
 	}
 
+	// try to load cred from configured credential store
```
Comment (dabooz) [inline cli/cliutils/cliutils.go:218]: Please put a comment on the GetDockerAuth function indicating that it also obtains creds from a docker credential store if it's in use.

File: cli/cliutils/cliutils.go [hunk 2]
```diff
@@ -214,6 +215,49 @@ func GetDockerAuth(domain string) (auth dockerclient.AuthConfiguration, err erro
...
+	var cfgPath string
+	if dockerCfgPath := os.Getenv("DOCKER_CONFIG"); dockerCfgPath != "" {
```
Comment (dabooz) [inline cli/cliutils/cliutils.go:220]: Why didnt you use the docker auth from cred store function in go-dockerclient? "https://github.com/fsouza/go-dockerclient/blob/main/auth.go#L296"

**Review by dabooz — CHANGES_REQUESTED** [review]

File: cli/cliutils/cliutils.go (hunk 2 above)
Comment (sirotsinskuy) [reply cli/cliutils/cliutils.go:220]: In the current version of go-dockerclient we're using this functionality isn't available. Can I update version to the latest - 1.7.2?

Comment (dabooz) [reply cli/cliutils/cliutils.go:220]: Yes, I think that would be a more appropriate way to address the problem.

**Review by dabooz — CHANGES_REQUESTED** [review]
Ok, this looks like it solves the problem for hzn exchange service publish when it has to manipulate the image, but what about the agent at runtime? Does the CLI code push the creds from the docker cred store to the exchange when the service is published? If so, I am not seeing it in the code.

**Review by dabooz — DISMISSED** [review]

**Review by linggao — APPROVED** [review]

**Pull request merged**